tests/functional/Project.php

```diff
@@ -496,4 +496,156 @@ public function testProjectStateColorInSearchResults(): void
         $this->integer(substr_count($html, "background-color: #000002;"))->isEqualTo(2);
         $this->integer(substr_count($html, "background-color: #000003;"))->isEqualTo(3);
     }
+
+    public function providerGetActiveProjectIDsForUser(): iterable
+    {
+        // Check if a user with no projects returns an empty array
+        yield ['user0', false, false, false, false, false, false, false, false, true];
```

Comment:
I appreciate that you are trying to test every possibilities but I am afraid the form of this provider is not ideal.
Indeed, using so many boolean parameters make it very hard to understand the test case that is being run.
This may lead to very efficient code but it costs too much readability.

In a perfect world, a test case should be super readable by containing only its parameters and the expected result.

In my opinion, here our parameters should only be:  
* The user
* One or two boolean depending on the tested method available parameters (search_in_groups, search_in_team, ...) 
* The expected projects id.

Adding key names is also needed because of these two boolean (otherwise it might to hard to remember which is which).

For example:
```php
yield [
    'user' => $user_without_projects, 
    'search_in_team' => false,
    'search_in_groups => false,
    'expected' => [] 
];
```

This require that `$user_without_projects` is set up "manually" before the `yield` statement (which may cause a bit of code duplication, but that can be minimized by using a dedicated method that can be reused by others similar tests).